planning/behavior_path_planner/autoware_behavior_path_lane_change_module/src/utils/utils.cpp

```diff
@@ -1323,4 +1286,68 @@ bool has_overtaking_turn_lane_object(
   return std::any_of(
     trailing_objects.begin(), trailing_objects.end(), is_object_overlap_with_target);
 }
+
+bool filter_target_lane_objects(
+  const CommonDataPtr & common_data_ptr, const ExtendedPredictedObject & object,
+  const double dist_ego_to_current_lanes_center, const bool ahead_of_ego,
+  const bool before_terminal, TargetLaneLeadingObjects & leading_objects,
+  ExtendedPredictedObjects & trailing_objects)
+{
+  using behavior_path_planner::utils::path_safety_checker::filter::velocity_filter;
+  const auto & current_lanes = common_data_ptr->lanes_ptr->current;
+  const auto & vehicle_width = common_data_ptr->bpp_param_ptr->vehicle_info.vehicle_width_m;
+  const auto & lanes_polygon = *common_data_ptr->lanes_polygon_ptr;
+  const auto stopped_obj_vel_th = common_data_ptr->lc_param_ptr->stopped_object_velocity_threshold;
+
+  const auto is_lateral_far = std::invoke([&]() -> bool {
```

Comment:
What is the purpose of this check?

Reply:
To compensate on perception error for rear object

See https://github.com/autowarefoundation/autoware.universe/pull/4930#:~:text=Using%20PSIM-,Before%20PR,-Unnecessary%20lane%20change